drodrpg/DRODLib/Monster.cpp

```diff
@@ -1555,15 +1555,16 @@ std::array<float, 3 > CMonster::getHSV() const
 UINT CMonster::getDEF() const
 //Return: monster's DEF
 {
-	if (IsOnMistTile() && !IsMistImmune())
-		return 0; //Mist tile nullifies DEF
-
-	UINT val = this->DEF;
+	int val = this->DEF;
 	if (!this->pCurrentGame || !val)
 		return val;
 	const float fMult = this->pCurrentGame->GetTotalStatModifier(ScriptVars::MonsterDEF);
-	val = UINTBounds(val * fMult);
-	return !val ? 1 : val; //DEF always >= 1
+	val = intBounds(val * fMult);
+
+	if (val < 0 && IsOnMistTile() && !IsMistImmune())
+		return 0; //Mist tile nullifies +ve DEF
```

Comment:
I think the "+ve" is a typo

Reply:
It's a shorthand but the comment isn't too long. The real typo is that I used the wrong comparator, so mist would stop nullifying positive ATK!